enhancements/baremetal-IPI/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,185 @@
+being added to a new CR used only in "metal3" deployments. This new CR would be
+within the operator scope and will only contain contain configuration required
+by the provisioning service to boot baremetal servers.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
+Update lifecycle for bootimage is an unsolved problem for all platforms. For
+more information, refer to : https://github.com/openshift/os/issues/381.
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "BareMetalController" or "Metal3Controller" would be created within
+the operator scope.(the Controller suffix seems inaccurate but other CRs in the
+operator scope seem to follow this pattern. Also, to make things clearer, we could
+call this CR, BareMetal/Metal3ProvisioningController.)
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying
+baremetal server which would be physically connected to the provisioning
+network. This configuration is needed only for the underlying
+provisioning service (Ironic) and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on
+the baremetal server for provisioning. This is also a value that useful
+just to the provisioning service. This value should not be in the DHCP
+range and should not be used in the provisioning network for any other
+purpose (should be a free IP address.) It is expected to be provided in
+the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its
+CIDR. The ProvisioningIP and the ProvisioningDHCPRange are expected to
+be within this network. The value for this config is expected to be
+provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which
+baremetal hosts can be assigned their IP address. This range should be
+provided as two comma seperated IP addresses where the addresses represent
+a contiguous range of IPs that are available to be allocated. It is expected
+to be provided in the format : 10.1.0.10, 10.1.0.100.
+
+5. CacheURL : This is a location where image required to bring up baremetal
+hosts have been previously downloaded for faster image downloads within the
+cluster. This is an optional parameter and is expected to be provided
+in the format : http://111.111.111.1/images
+
+6. RHCOSImageURL : This config is expected to be the location of the
```

Comment:
We should make clear that we expect this URL to point to a server outside of the cluster somewhere.

Reply:
Following recent discussions one option could be to remove both RHCOSImageURL and CacheURL, and disable/remove the hosted caching completely.  Then we'd rely on externally managed caching and point the machine providerspec at that external cache.  This could simplify things somewhat, particularly the update lifecycle of the cached images.

Reply:
Ok. Based on our recent discussions, removing the RHCOSImageURL and the CacheURL from this CR. At this point, we will require the user to provide an externally managed cache and the Machine Provider Spec will point to that cache URL. 
No information about an image would be provided to Metal3 during deployment.